lib/twilio/audiohelper.ts

```diff
@@ -780,7 +788,14 @@ class AudioHelper extends EventEmitter {
    * @param forceGetUserMedia - If true, getUserMedia will be called even if
    *   the specified device is already active.
    */
-  private _setInputDevice(deviceId: string, forceGetUserMedia: boolean): Promise<void> {
+  private async _setInputDevice(deviceId: string, forceGetUserMedia: boolean): Promise<void> {
+    const connectPromise = this._getConnectPromise();
+    if (connectPromise) {
+      this._log.debug('connectPromise detected, waiting...');
+      await connectPromise;
+      this._log.debug('connectPromise resolved');
+    }
+
```

Comment:
If you add a default, you can change this to one line
```
await this._getConnectPromise();
```

Reply:
Do we no longer want the debug lines? I see we have debug lines https://github.com/twilio/twilio-voice.js/blob/master/lib/twilio/device.ts#L1025

Reply:
We can add the log in the constructor where we assign the default.

Reply:
Without the null, not sure where would be best to place this since a conditional for `options.getMakeCallPromise` will always return true. Maybe better to put the debug lines in `_makeCallPromise`?